cabal-install/Distribution/Client/ProjectOrchestration.hs

```diff
@@ -342,16 +365,18 @@ resolveAndCheckTargets targetDefaultComponents
       = Left (BuildTargetNotInProject (buildTargetPackage t))
 
 
-    projAllPkgs, projLocalPkgs :: Map PackageName InstalledPackageId
+    -- NB: Make sure you get the *un-munged* PackageName
+    projAllPkgs, projLocalPkgs :: Map PackageName [InstalledPackageId]
```

Comment:
Is this fixing an existing problem (that we can have multiple instances of the same package name) or is there something new in this patch that makes this an issue that has to be addressed?